src/spi.rs

```diff
@@ -277,6 +281,22 @@ pub mod config {
             self
         }
 
+        /// Add an aditional Amount of SPI bit-cycles the cs should be activated before the transmission (0-16).
+        /// This only works on half-duplex transactions.
+        #[must_use]
+        pub fn cs_pre_delay_us(mut self, delay_us: u16) -> Self {
```

Comment:
How does it behave when not in half-duplex mode? Do we get a runtime EspError or does it just get eaten silently? If you are not sure could you test out the cases and report how it acts running?

Reply:
That's a very funny thing in fact. I only use the FullDuplex mode.
I am using the AI-Thinker NodeMcu ESP32-S.
And it does make a difference (small one, but there is).  However it only seems to take in account the fact that its different from 0.
When I do not set it or put 0, we have no time gap between CS pull-down and the first clock (and you can see the turquoise MISO channel doesn't change its state):

- 0 ![cs_pre_delay_us_0](https://github.com/esp-rs/esp-idf-hal/assets/75721408/f5eee3a6-7bc3-4785-bd63-65d8db991610)

When I set it to 1/16/1000, I get the same small time gap between the CS pull-down and the first clock (and MISO answers):
- 1 ![cs_pre_delay_us_1](https://github.com/esp-rs/esp-idf-hal/assets/75721408/918f1762-3220-45b1-afc8-152d84bf5eb5)
- 16 ![cs_pre_delay_us_16](https://github.com/esp-rs/esp-idf-hal/assets/75721408/cf40e640-6cf6-47cd-8901-1330945219bc)
- 1000 ![cs_pre_delay_us_1000](https://github.com/esp-rs/esp-idf-hal/assets/75721408/35e50f84-9efc-43f6-9b67-93d6d17dfc70)

This is all using FullDuplex.

They say in the C API driver include that its not supported, but somewhere else if we look closely we see:
![image](https://github.com/esp-rs/esp-idf-hal/assets/75721408/4c743718-d552-4068-ba50-982e11f41407)

> In full-duplex mode, only support cs pretrans delay = 1 and without address_bits and command_bits"

So, at least from what I see its that, even though some part of the C doc seems to say that in FullDuplex mode it doesn't work. In reality its more like a on-off switch for a 1us pre-delay in FullDuplex mode, but it does make a big difference for some SPI chips.

Also, as you can see, setting a value higher than 16 (here 1000), doesn't change the behavior compared to 16.

(And sorry for the bad pictures instead of the oscilloscope screen capture, I was lazy to remember the functionality, but I think its still is enough to demonstrate)

Reply:
Further reading of the [doc](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/spi_master.html#spi-known-issues) says:
> `cs_ena_pretrans` is not compatible with the Command and Address phases of full-duplex transactions.

Reply:
Thanks for testing it out!
We are not using the Command & Address part currently in the driver. So that bit doesn't matter. 
Did you also test the post delay? Docs don't indicate that this isn't only working in Half-Duplex. 

What version of esp-idf are you using? We could annotate your finding's that a delay is happening in Full-Duplex at least in your tested esp-idf version, but with a warning that it might not be a "stable" option to do so.

Reply:
This week will be busy at work for reports etc... I will see if I have time, if not it will be next week!

Reply:
Its your PR -> your pacing :smile: 
i have no stress here